lib/Conversion/TorchToMhlo/BasicOp.cpp

```diff
@@ -25,6 +26,470 @@ using namespace mlir;
+  LogicalResult
+  matchAndRewrite(AtenOpT op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    auto outType = OpConversionPattern<AtenOpT>::getTypeConverter()
+                       ->convertType(op.getType())
+                       .template dyn_cast<TensorType>();
+
+    if (!outType)
+      return op.emitError("Only Tensor types supported in MHLO");
+
+    Type outElemTy = outType.getElementType();
+    if (!outElemTy.isIntOrFloat())
+      return op.emitError(
+          "Only floating-point or integer datatype legalization supported");
+
+    // FIXME: Handle layout, device and pin_memory. Assume dtype has been
+    // processed to set output type correctly?
+    if (!op.layout().getType().template isa<Torch::NoneType>())
+      return op.emitError("Only default layout is supported");
+
+    bool pinMemory;
+    if (!op.pin_memory().getType().template isa<Torch::NoneType>() &&
+        (!matchPattern(op.pin_memory(), m_TorchConstantBool(&pinMemory)) ||
+         pinMemory)) {
+      return op.emitError(
+          "Unsupported pin_memory, should be either None or false");
+    }
+
+    SmallVector<int64_t> shape;
+    if (!matchPattern(op.size(), m_TorchConstantIntList(shape))) {
+      return op.emitError("Shape must be a list of Scalar constants");
+    }
+
+    int64_t size = 1;
+    for (auto s : shape)
+      size *= s;
+
+    SmallVector<int32_t> values(size, fillVal);
+    auto constOp =
+        mhlo::getConstTensor<int32_t>(rewriter, op, values, shape).getValue();
+
+    rewriter.replaceOpWithNewOp<mhlo::ConvertOp>(op, outType, constOp);
+    return success();
+  }
+};
+
+} // namespace
+
+// These binary op legalizations are specific to add/sub which have an
+// alpha multiplier.
+namespace {
+template <typename AtenOpT, typename MhloOpT>
+class ConvertAtenAddSubOp : public OpConversionPattern<AtenOpT> {
+public:
+  using OpConversionPattern<AtenOpT>::OpConversionPattern;
+  using OpAdaptor = typename AtenOpT::Adaptor;
+  LogicalResult
+  matchAndRewrite(AtenOpT op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Value lhs = adaptor.self();
+    TensorType lhsType = lhs.getType().dyn_cast<TensorType>();
+    Value rhs = adaptor.other();
+    TensorType rhsType = rhs.getType().dyn_cast<TensorType>();
+
+    if (!lhsType)
+      return op.emitError("Only Tensor types supported in MHLO");
+
+    TensorType outType = OpConversionPattern<AtenOpT>::getTypeConverter()
+                             ->convertType(op.getType())
+                             .template cast<TensorType>();
+
+    Type outElemTy = outType.getElementType();
+    if (!outElemTy.isIntOrFloat()) {
+      return op.emitError(
+          "Only floating-point or integer datatype legalization supported");
+    }
+
+    Value rhsAsTensor;
+    if (!rhsType) {
+      if (failed(mhlo::torchScalarToMhloTensor(rewriter, op, op.other(),
```

Comment:
Nit: Can we extend the `torchScalarToMhloTensor` to convert also scalar variable but not only constant? Because `add/sub/mul/div` the second operand is not guaranteed to be a constant.

Reply:
Yes. Supporting for variable alpha or none-constant rhs in add/sub/mul/div will introduce many code changes. Since it's just a setup, we'd like to support it later.